llama_index/indices/multi_modal/base.py

```diff
@@ -0,0 +1,329 @@
+"""Multi Modal Vector Store Index.
+
+An index that that is built on top of multiple vector stores for different modalities.
+
+"""
+import logging
+from typing import Any, List, Optional, Sequence
+
+from llama_index.data_structs.data_structs import IndexDict
+from llama_index.embeddings.mutli_modal_base import MultiModalEmbedding
+from llama_index.embeddings.utils import EmbedType, resolve_embed_model
+from llama_index.indices.base_retriever import BaseRetriever
+from llama_index.indices.service_context import ServiceContext
+from llama_index.indices.utils import (
+    async_embed_image_nodes,
+    async_embed_nodes,
+    embed_image_nodes,
+    embed_nodes,
+)
+from llama_index.indices.vector_store.base import VectorStoreIndex
+from llama_index.schema import BaseNode, ImageNode, IndexNode
+from llama_index.storage.storage_context import StorageContext
+from llama_index.vector_stores.simple import DEFAULT_VECTOR_STORE, SimpleVectorStore
+from llama_index.vector_stores.types import VectorStore
+
+logger = logging.getLogger(__name__)
+
+
+class MultiModalVectorStoreIndex(VectorStoreIndex):
+    """Multi-Modal Vector Store Index.
+
+    Args:
+        use_async (bool): Whether to use asynchronous calls. Defaults to False.
+        show_progress (bool): Whether to show tqdm progress bars. Defaults to False.
+        store_nodes_override (bool): set to True to always store Node objects in index
+            store and document store even if vector store keeps text. Defaults to False
+    """
+
+    image_store_key = "image"
+
+    def __init__(
+        self,
+        nodes: Optional[Sequence[BaseNode]] = None,
+        index_struct: Optional[IndexDict] = None,
+        service_context: Optional[ServiceContext] = None,
+        storage_context: Optional[StorageContext] = None,
+        use_async: bool = False,
+        store_nodes_override: bool = False,
+        show_progress: bool = False,
+        # Image-related kwargs
+        image_vector_store: Optional[VectorStore] = None,
+        image_embed_model: EmbedType = "clip",
```

Comment:
one thing I am not sure here is that should we put "default" here. And map default to Clip emb in util.py.
but this way should be fine.

if we have future new image embedding models, we need to change clip everywhere

Reply:
I think that's only a concern if we change the default away from clip (which tbh, we should try not to change the default, especially for embeddings)

Reply:
kk got it